src/main/java/org/springframework/cloud/deployer/spi/kubernetes/AbstractKubernetesDeployer.java

```diff
@@ -413,21 +413,31 @@ protected boolean getHostNetwork(AppDeploymentRequest request) {
 	/**
 	 * Get the nodeSelectors setting for the deployment request.
 	 *
-	 * @param properties The deployment request deployment properties.
+	 * @param deploymentProperties The deployment request deployment properties.
 	 * @return map of nodeSelectors
 	 */
-	protected Map<String, String> getNodeSelectors(Map<String, String> properties) {
+	private Map<String, String> getNodeSelectors(Map<String, String> deploymentProperties) {
 		Map<String, String> nodeSelectors = new HashMap<>();
 
-		String nodeSelectorsProperty = properties
+		String nodeSelector = properties.getNodeSelector();
+
+		String nodeSelectorDeploymentProperty = deploymentProperties
 				.getOrDefault(KubernetesDeployerProperties.KUBERNETES_DEPLOYMENT_NODE_SELECTOR, "");
 
-		if (StringUtils.hasText(nodeSelectorsProperty)) {
-			String[] nodeSelectorPairs = nodeSelectorsProperty.split(",");
+		boolean hasGlobalNodeSelector = StringUtils.hasText(properties.getNodeSelector());
+		boolean hasDeployerPropertyNodeSelector = StringUtils.hasText(nodeSelectorDeploymentProperty);
```

Comment:
`hasDeployerPropertyNodeSelector` -> is a bit misleading. It is not a deployer property, though, but rather it is a "per deployment-level" property ... so let's revise it.